File: Resolução em python/ex5.py
total_sal = 0
qtd_nutri = 0

while True:
  cod = int(
      input(
          "Informe o código do cargo (1 - Enfermeiro, 2 - Nutricionista, 3 - Médico): "
      ))

  if cod == 0:
    break

  salario = float(input("Informe o salário do funcionário: "))

  if cod == 2:
    total_sal += salario
    qtd_nutri += 1

if qtd_nutri > 0:
  media_sal = total_sal / qtd_nutri
  print("A média salarial dos nutricionistas é: R$ {:.2f}".format(media_sal))
else:
  print("Não foram informados salários de nutricionistas.")

File: Resolução em C/ex5.c
#include <stdio.h>

int ex4() {
  int codigoCargo;
  float salario;
  int cont = 0;
  float somaSal = 0;

  // Loop para leitura dos dados
  while (1) {
    // Entrada de dados
    printf("1- Enfermeiro\n2- Nutricionista\n3- Médico\nInforme o código do "
           "cargo (0 para encerrar): ");
    scanf("%d", &codigoCargo);

    // Condição de parada
    if (codigoCargo == 0) {
      break;
    }

    // Entrada de dados
    printf("Informe o salário: ");
    scanf("%f", &salario);

    // Verificação do cargo e cálculo da média salarial dos nutricionistas
    if (codigoCargo == 2) {
      cont++;
      somaSal += salario;
    }
  }

  // Cálculo da média salarial dos nutricionistas
  float mediaSal = 0;
  if (cont > 0) {
    mediaSal = somaSal / cont;
  }

  // Exibição do resultado
  printf("A média salarial dos nutricionistas é: %.2f\n", mediaSal);

  return 0;
}